### ISSUE TITLE
retina-agent pod initialization failed to no BTF found for kernel version

### ISSUE THREAD

**Issue by Radiance2024** [microsoft/retina#247]
**Describe the bug**
- installation commands: 
make helm-install-with-operator
- retina-agent pod status as follows:
`kube-system            pod/retina-agent-bq6s8                          0/1     CrashLoopBackOff   5 (2m29s ago)`
`kube-system            pod/retina-agent-lkpgn                          0/1     CrashLoopBackOff   5 (2m23s ago)`
`kube-system            pod/retina-agent-psqgp                          0/1     CrashLoopBackOff   5 (2m29s ago)`
`kube-system            pod/retina-operator-8445c8c6f6-st5mq            1/1     Running            0 `

- retina-agent containers error logs is:
ts=2024-04-09T09:19:38.229Z level=info caller=server/server.go:71 msg="HTTP server stopped with err: http: Server closed" goversion=go1.21.8 os=linux arch=amd64 numcores=1 hostname=k8s-node2 podname=retina-agent-bq6s8 version=v0.0.4 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns
ts=2024-04-09T09:19:38.229Z level=panic caller=controllermanager/controllermanager.go:119 msg="Error running controller manager" goversion=go1.21.8 os=linux arch=amd64 numcores=1 hostname=k8s-node2 podname=retina-agent-bq6s8 version=v0.0.4 apiserver=https://10.96.0.1:443 plugins=dropreason,packetforward,linuxutil,dns error="failed to reconcile plugin dropreason: field InetCskAccept: program inet_csk_accept: apply CO-RE relocations: load kernel spec: no BTF found for kernel version 5.4.272-1.el7.elrepo.x86_64: not supported" errorVerbose="field InetCskAccept: program inet_csk_accept: apply CO-RE relocations: load kernel spec: no BTF found for kernel version 5.4.272-1.el7.elrepo.x86_64: not supported\nfailed to reconcile plugin dropreason\ngithub.com/microsoft/retina/pkg/managers/pluginmanager.(*PluginManager).Start\n\t/go/src/github.com/microsoft/retina/pkg/managers/pluginmanager/pluginmanager.go:169\ngithub.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start.func1\n\t/go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:109\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\t/go/pkg/mod/golang.org/x/sync@v0.6.0/errgroup/errgroup.go:78\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1650"
panic: Error running controller manager

goroutine 145 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x1?, 0x41a508?, {0x0?, 0x0?, 0xc000a05940?})
        /go/pkg/mod/go.uber.org/zap@v1.26.0/zapcore/entry.go:196 +0x54
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0008f0c30, {0xc000830300, 0x1, 0x1})
        /go/pkg/mod/go.uber.org/zap@v1.26.0/zapcore/entry.go:262 +0x3ec
go.uber.org/zap.(*Logger).Panic(0xc000830000?, {0x2b52d30?, 0x8fbec0?}, {0xc000830300, 0x1, 0x1})
        /go/pkg/mod/go.uber.org/zap@v1.26.0/logger.go:284 +0x51
github.com/microsoft/retina/pkg/managers/controllermanager.(*Controller).Start(0xc000615ef0, {0x2f10a90?, 0xc000615ea0?})
        /go/src/github.com/microsoft/retina/pkg/managers/controllermanager/controllermanager.go:119 +0x28c
created by main.main in goroutine 1
        /go/src/github.com/microsoft/retina/controller/main.go:290 +0x28d0

**Expected behavior**
- retina-agent pod status is normal.

**Platform (please complete the following information):**

- OS: CentOS Linux 7 (Core)
- Kernel Version: 5.4.272-1.el7.elrepo.x86_64
- Kubernetes Version: 1.24.9
- Host: local kubernets
- Retina Version: v0.0.4

**Comment by rbtr** [comment]
Reopened this to triage it with the team. Given CentOS 7 doesn't really support eBPF/BTF and will anyway be [EOL in 2 months](https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/), we are closing this as wontfix.

**Comment by Radiance2024** [comment]
@rbtr Thank you for your response. We found that the effectiveness of enabling BTF support on CentOS 7 through the BTFHub installation method is not satisfactory. We achieved eBPF support on CentOS 7 by upgrading the kernel version, but also discovered that BTF is not supported by default. Many of our servers are still running on CentOS 7, and we have also observed the lack of default BTF support on servers using the aarch architecture. We would like to suggest that Retina consider providing compatibility support for systems that do not have BTF support enabled by default. Thank you.
